solidity/test/submitERC721LogicCall.ts

```diff
@@ -0,0 +1,448 @@
+import chai from "chai";
+import { ethers } from "hardhat";
+import { solidity } from "ethereum-waffle";
+import { TestERC721A } from "../typechain/TestERC721A";
+import { GravityERC721 } from "../typechain/GravityERC721";
+import { deployContracts } from "../test-utils/deployERC721";
+import {
+  getSignerAddresses,
+  signHash,
+  examplePowers,
+  ZeroAddress
+} from "../test-utils/pure";
+import { ERC721, TestERC20A } from "../typechain";
+
+chai.use(solidity);
+const { expect } = chai;
+
+
+async function runTest(opts: {
+  // Issues with the tx batch
+  invalidationNonceNotHigher?: boolean;
+  malformedTxBatch?: boolean;
+
+  // Issues with the current valset and signatures
+  nonMatchingCurrentValset?: boolean;
+  badValidatorSig?: boolean;
+  zeroedValidatorSig?: boolean;
```

Comment:
in this case all of these edge cases are duplicates from the generic logic call test, we can remove them and replace them with only a test for not having the correct NFT ID or contract.

Reply:
Removed all the edge cases that were duplicated in the other test. Now have three tests: 
1) happy path
2) throw on wrong caller of gravityERC721
3) throw on wrong owner of ERC721 token